### DISCUSSION TITLE
Should each style change be a separate render pass?

### DISCUSSION THREAD

**Discussion by rokit** [grovesNL/glyphon#56] [Q&A] [answered]
For example, `buffer.set_text` with `Style::Italic`, then `buffer.set_text` with `Style::Normal` would be two render passes?

**Comment by grovesNL** [comment, marked as answer]
One render pass that renders multiple buffers is ok, and one buffer can have multiple styles. You can check the cosmic-text examples to see how to apply styles to specific spans within a buffer. Let me know if you have any trouble applying it like this.

In general you should try to share a single render pass as much as possible.

**Reply by rokit** [reply]
I checked Cosmic Text for multiple spans, but it uses [`set_rich_text`](https://github.com/pop-os/cosmic-text/blob/3f84eb7cb4c03f71bda9e106617b9e275e873bde/examples/rich-text/src/main.rs#L122), which I don't see in this library. Could you add an example for that case?
